### DISCUSSION TITLE
Avalonia equivalent of a FrameworkContentElement?

### DISCUSSION THREAD

**Discussion by Kevin-Seiden** [AvaloniaUI/Avalonia#11254] [Q&A] [answered]
I'm trying to port a WPF custom control to Avalonia which has a pair of ObservableCollections (AvaloniaLists?) which act as content for the control.  In this case the content can not render itself, it relies on the parent to render it, but are still added as logical children of the control.  In WPF I inherit FrameworkContentElement to create these, what would I inherit in Avalonia to get similar effect?  I could create them as TemplatedControls but their templates wouldn't be used so I'm just wondering if there is something less overkill for the purpose.

**Comment by Gillibald** [comment, marked as answer]
StyledElement gives access to logical and visual tree

**Reply by Kevin-Seiden** [reply]
Thank you very much.  That worked perfectly and I was able to get my control completely functional.
